specs/core/0_beacon-chain.md

```diff
@@ -198,32 +198,30 @@ The following values are (non-configurable) constants used throughout the specif
 | - | - |
 | `GENESIS_SLOT` | `Slot(0)` |
 | `GENESIS_EPOCH` | `Epoch(0)` |
-| `BLS_WITHDRAWAL_PREFIX` | `0` |
+| `BLS_WITHDRAWAL_PREFIX` | `Bytes1()` |
 
 ### Time parameters
 
-| Name | Value | Unit | Duration |
+| Name | Value | Duration |
 | - | - | :-: | :-: |
-| `MIN_ATTESTATION_INCLUSION_DELAY` | `2**0` (= 1) | slots | 6 seconds |
-| `SLOTS_PER_EPOCH` | `2**6` (= 64) | slots | 6.4 minutes |
-| `MIN_SEED_LOOKAHEAD` | `2**0` (= 1) | epochs | 6.4 minutes |
-| `ACTIVATION_EXIT_DELAY` | `2**2` (= 4) | epochs | 25.6 minutes |
-| `SLOTS_PER_ETH1_VOTING_PERIOD` | `2**10` (= 1,024) | slots | ~1.7 hours |
-| `SLOTS_PER_HISTORICAL_ROOT` | `2**13` (= 8,192) | slots | ~13 hours |
-| `MIN_VALIDATOR_WITHDRAWABILITY_DELAY` | `2**8` (= 256) | epochs | ~27 hours |
-| `PERSISTENT_COMMITTEE_PERIOD` | `2**11` (= 2,048)  | epochs | 9 days  |
-| `MAX_EPOCHS_PER_CROSSLINK` | `2**6` (= 64) | epochs | ~7 hours |
-| `MIN_EPOCHS_TO_INACTIVITY_PENALTY` | `2**2` (= 4) | epochs | 25.6 minutes |
-
-* `MAX_EPOCHS_PER_CROSSLINK` should be a small constant times `SHARD_COUNT // SLOTS_PER_EPOCH`.
+| `MIN_ATTESTATION_INCLUSION_DELAY` | `Slot(2**0)` (= 1) | 6 seconds |
+| `SLOTS_PER_EPOCH` | `Slot(2**6)` (= 64) | 6.4 minutes |
+| `MIN_SEED_LOOKAHEAD` | `Epoch(2**0)` (= 1) | 6.4 minutes |
+| `ACTIVATION_EXIT_DELAY` | `Epoch(2**2)` (= 4) | 25.6 minutes |
+| `SLOTS_PER_ETH1_VOTING_PERIOD` | `Slot(2**10)` (= 1,024) | ~1.7 hours |
+| `SLOTS_PER_HISTORICAL_ROOT` | `Slot(2**13)` (= 8,192) | ~13 hours |
+| `MIN_VALIDATOR_WITHDRAWABILITY_DELAY` | `Epoch(2**8)` (= 256) | ~27 hours |
+| `PERSISTENT_COMMITTEE_PERIOD` | `Epoch(2**11)` (= 2,048) | 9 days |
+| `MAX_EPOCHS_PER_CROSSLINK` | `Epoch(2**6)` (= 64) | ~7 hours |
+| `MIN_EPOCHS_TO_INACTIVITY_PENALTY` | `Epoch(2**2)` (= 4) | 25.6 minutes |
 
 ### State list lengths
 
-| Name | Value | Unit | Duration |
+| Name | Value | Duration |
 | - | - | :-: | :-: |
-| `EPOCHS_PER_HISTORICAL_VECTOR` | `2**16` (= 65,536) | epochs | ~0.8 years |
-| `EPOCHS_PER_SLASHINGS_VECTOR` | `2**13` (= 8,192) | epochs | ~36 days |
-| `HISTORICAL_ROOTS_LIMIT` | `2**24` (= 16,777,216) | historical roots | ~26,131 years |
+| `EPOCHS_PER_HISTORICAL_VECTOR` | `Epoch(2**16)` (= 65,536) | ~0.8 years |
```

Comment:
Ditto

Reply:
I think I disagree:

* The name `EPOCHS_PER_...` is consistent with the `Epoch` type.
* `EPOCHS_PER_HISTORICAL_VECTOR` naturally occurs in `Epoch` summations such as `epoch + EPOCHS_PER_HISTORICAL_VECTOR - MIN_SEED_LOOKAHEAD`.
* `EPOCHS_PER_HISTORICAL_VECTOR` naturally occurs in `Epoch` modular reductions such as `epoch % EPOCHS_PER_HISTORICAL_VECTOR`.

Reply:
I think in instances where there is a `PER` in the name, dimensional analysis dictates that they are not `Epochs`, but `Epochs * HistoricalVectors**(-1).

Reply:
`SLOTS_PER_EPOCH` is defined as "the number of slots in an epoch". That's of type `Slot`, not `Slot/Epoch`.

Reply:
Disagree. The use of `SLOTS_PER_EPOCH` in `slot_to_epoch` and in `epoch_start_slot` stand counter to your argument.

If `SLOTS_PER_EPOCH` is a `Slot` then `epoch_start_slot`'s return type  is `EpochSlot` and `epoch_start_slot` is dimensionless.

Reply:
Historical vector here doesn't represent a type, but a non-typed *scalar*. So you could say we are multiplying by something that doesn't change the type, only the length. Imho I think the debate is silly, and naming is consistent and clear.

Reply:
Hum, OK. Would you agree that `Epochs * HistoricalVectors**(-1)` is the same as `Epochs` because `HistoricalVectors` is dimensionless? 😂

Reply:
I don't care too much either way. Need to go to sleep soon as I can't think straight anymore 💤

Reply:
😂Fine. :) I am not entirely convinced, but I sort of agree enough on the dimensionlessness `HistoricalVectors`.

Reply:
Agreed the need for sleep is real.